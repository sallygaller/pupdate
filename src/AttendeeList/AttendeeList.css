.AttendeeList {
   margin: 0 auto;
}

.AttendeeList-hidden {
    display: none;
}

.AttendeeList-play-profile {
    background-color: #0081a7;
    color: white;
    margin: 0px 15px;
}

li {
    padding: 10px;
}