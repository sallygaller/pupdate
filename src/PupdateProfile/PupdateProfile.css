.PupdateProfile {
    text-align: center;
    width: 50%;
    margin: 0 auto;
}

.PupdateProfile ul, li {
    list-style-type: none;
    list-style-position: inside;
    margin: 0;
    padding: 0; 
}
.PupdateProfile div {
    margin-bottom: 15px;
}

.PupdateProfile button {
    padding: 10px;
    border: #F2AD1D solid 2px;
    border-radius: 25px;
    cursor: pointer;
    padding: 10px;
    text-align: center;
}

.PupdateProfile-attendee {
    margin: 0px;
}

.PupdateProfile button:hover {
    background-color: #F2AD1D;
}

.PupdateProfile a,
.PupdateProfile a:active,
.PupdateProfile a:focus,
.PupdateProfile a:visited {
  color: black;
  font-weight: bold;
  text-decoration: none;
}

.PupdateProfile-item {
    text-align: left;
}

.PupdateProfile-edit {
    margin-right: 25px;
}