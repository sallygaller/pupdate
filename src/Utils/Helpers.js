export default function playstyles(playstyle) {
  if (playstyle === "nervous") {
    return "I'm nervous or shy around other dogs.";
  }
  if (playstyle === "gentle") {
    return "I play gently.";
  }
  if (playstyle === "food-obsessed") {
    return "I'm food-obsessed!";
  }
  if (playstyle === "walks") {
    return "I like going on walks with my pup pals.";
  }
  if (playstyle === "parks") {
    return "I like going to dog parks with my pup pals.";
  }
  if (playstyle === "wrestling") {
    return "I like playfighting and wrestling.";
  }
  if (playstyle === "rambunctious") {
    return "I'm rambunctious and playful.";
  }
  if (playstyle === "ball-obsessed") {
    return "I love to play fetch!";
  }
}
