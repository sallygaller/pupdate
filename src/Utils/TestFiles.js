export const pupdates = [
  {
    id: 1,
    date: "2020-03-02T08:00:00.000Z",
    starttime: "09:30:00",
    endtime: "10:30:00",
    location: "Arbor Lodge Park",
    description: "Let's hang out at the park!",
    organizer: 1,
  },
  {
    id: 2,
    date: "2020-03-01T08:00:00.000Z",
    starttime: "15:30:00",
    endtime: "17:30:00",
    location: "Arbor Lodge Park",
    description: "Let's hang out at the park! I'll bring a frisbee",
    organizer: 1,
  },
];

export const pups = [
  {
    id: 1,
    name: "Eddie",
    breed: "dachshund",
    mix: true,
    age: "Adult",
    size: "M",
    nervous: true,
    rambunctious: false,
    gentle: true,
    wrestling: false,
    walks: true,
    parks: false,
    foodobsessed: true,
    ballobsessed: false,
    description:
      "Eddie can be a bit shy around other dogs at first, but he is really sweet! He loves going on long walks where there is a lot to sniff",
    date_added: "2020-12-25T09:28:32.615Z",
    owner: 1,
  },
  {
    id: 2,
    name: "Buddy",
    breed: "beagle",
    mix: true,
    age: "Adult",
    size: "L",
    nervous: false,
    rambunctious: true,
    gentle: false,
    wrestling: true,
    walks: false,
    parks: true,
    foodobsessed: false,
    ballobsessed: false,
    description:
      "Buddy can be a bit full on for some dogs, but he really just loves to play! He especially loves puppies",
    date_added: "2020-12-25T09:28:32.615Z",
    owner: 2,
  },
];
