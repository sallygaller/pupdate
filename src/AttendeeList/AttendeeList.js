import React from "react";
import { Link } from "react-router-dom";
import "./AttendeeList.css";

export default function AttendeeList(props) {
  if (props.showAttendees === false) {
    console.log("false");
  }
  return (
    <ul
      className={props.showAttendees === false ? "AttendeeList-hidden" : null}
    >
      {props.attendees.map((attendee) => {
        return (
          <li key={attendee.id}>
            {attendee.name}
            <Link to={`/pups/${attendee.id}`}>
              <button className="Pupdate-play-profile" type="button">
                View Play Profile
              </button>
            </Link>
          </li>
        );
      })}
    </ul>
  );
}
