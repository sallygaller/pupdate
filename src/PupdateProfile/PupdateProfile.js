import React from "react";
import { Link } from "react-router-dom";
import AttendeeList from "../AttendeeList/AttendeeList";
import PropTypes from "prop-types";
import moment from "moment";
import { API_ENDPOINT } from "../config";
import TokenService from "../services/token-service";
import "./PupdateProfile.css";

class PupdateProfile extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      pupdate: [],
      attendees: "",
      organizerPups: [],
      showAttendees: false,
      rsvp: "",
      userPupdate: false,
      userAttending: false,
      error: null,
    };
  }

  static propTypes = {
    match: PropTypes.shape({
      params: PropTypes.object,
    }),
    // history: PropTypes.shape({
    //   push: PropTypes.func,
    // }).isRequired,
  };

  componentDidMount() {
    const { pupdateId } = this.props.match.params;
    Promise.all([
      // get details about pupdate
      fetch(API_ENDPOINT + `/pupdates/${pupdateId}`, {
        headers: {
          authorization: `bearer ${TokenService.getAuthToken()}`,
        },
      }),
      // get rsvps for pupdate
      fetch(API_ENDPOINT + `/pupdate-rsvp/${pupdateId}`, {
        headers: {
          authorization: `bearer ${TokenService.getAuthToken()}`,
        },
      }),
      // get user organized pupdates
      fetch(API_ENDPOINT + `/pupdates/user`, {
        headers: {
          authorization: `bearer ${TokenService.getAuthToken()}`,
        },
      }),
      // get user rsvps
      fetch(API_ENDPOINT + `/pupdate-rsvp/user`, {
        headers: {
          authorization: `bearer ${TokenService.getAuthToken()}`,
        },
      }),
    ])
      .then(([res1, res2, res3, res4]) =>
        Promise.all([res1.json(), res2.json(), res3.json(), res4.json()])
      )
      .then(([responseData1, responseData2, responseData3, responseData4]) => {
        // set pupdate to state
        this.setState({
          pupdate: responseData1,
          rsvps: responseData2,
        });
        // check if user is attending this pupdate
        responseData4.map((pupdateRsvps) => {
          if (pupdateRsvps.pupdate === responseData1.id) {
            this.setState({
              userAttending: true,
              rsvp: pupdateRsvps.id,
            });
          }
        });
        // check if user is organizing this pupdate
        responseData3.map((pupdate) => {
          if (pupdate.id === responseData1.id) {
            this.setState({
              userPupdate: true,
            });
          }
        });
        return responseData1;
      })
      .then((responseData1) =>
        fetch(API_ENDPOINT + `/pups/user/${responseData1.id}`, {
          headers: {
            authorization: `bearer ${TokenService.getAuthToken()}`,
          },
        })
      )
      .then((res) => res.json())
      .then((responseData) =>
        this.setState({
          organizerPups: responseData,
        })
      )
      .catch((error) => {
        console.error({ error });
      });
  }

  handleAttendees = () => {
    if (this.state.showAttendees === false) {
      Promise.all(
        this.state.rsvps.map((pupdateRsvp) =>
          fetch(API_ENDPOINT + `/pups/user/${pupdateRsvp.attendee}`, {
            headers: {
              authorization: `bearer ${TokenService.getAuthToken()}`,
            },
          })
        )
      )
        .then((responses) => {
          return Promise.all(
            responses.map((response) => {
              return response.json();
            })
          );
        })
        .then((data) => {
          this.setState({
            attendees: data,
            showAttendees: true,
          });
        })
        .catch((error) => {
          console.error({ error });
        });
    } else {
      this.setState({
        showAttendees: false,
      });
    }
  };

  handleRsvpNo = (e) => {
    fetch(API_ENDPOINT + `/pupdate-rsvp/user/${this.state.rsvp}`, {
      method: "DELETE",
      headers: {
        "content-type": "application/json",
        authorization: `bearer ${TokenService.getAuthToken()}`,
      },
    })
      .then((res) => {
        if (!res.ok) return res.json().then((error) => Promise.reject(error));
        return res;
      })
      .then((data) => {
        this.props.history.push("/pupdates");
      })
      .catch((error) => {
        console.error({ error });
      });
  };

  handleRsvpYes = (e) => {
    fetch(API_ENDPOINT + `/pupdate-rsvp/${this.state.pupdate.id}`, {
      method: "POST",
      headers: {
        authorization: `bearer ${TokenService.getAuthToken()}`,
      },
    })
      .then((res) => {
        if (!res.ok) {
          return res.json().then((error) => {
            throw error;
          });
        }
        return res.json();
      })
      .then((data) => {
        this.props.history.push("/pupdates");
      })
      .catch((error) => {
        console.error({ error });
      });
  };

  handleBack = () => {
    this.props.history.goBack();
  };

  handleDeleteRequest = () => {
    fetch(API_ENDPOINT + `/pupdates/${this.state.pupdate.id}`, {
      method: "DELETE",
      headers: {
        "content-type": "application/json",
        authorization: `bearer ${TokenService.getAuthToken()}`,
      },
    })
      .then((res) => {
        if (!res.ok) return res.json().then((error) => Promise.reject(error));
        return res;
      })
      .then((data) => {
        this.props.history.push("/pupdates");
      })
      .catch((error) => {
        console.error({ error });
      });
  };

  render() {
    return (
      <div className="PupdateProfile">
        <section>
          <h2>Pupdate on {moment(this.state.pupdate.date).format("LL")}</h2>
          <div>
            <p>
              Time:{" "}
              {moment(this.state.pupdate.starttime, "h:mm A").format("h:mm A")}{" "}
              - {moment(this.state.pupdate.endtime, "h:mm A").format("h:mm A")}{" "}
              <br></br>
              Location: {this.state.pupdate.location}
            </p>
          </div>
          {this.state.userPupdate === true ? null : (
            <div>
              <h3>Organized by:</h3>
              <ul>
                {this.state.organizerPups.map((organizerPup) => (
                  <li key={organizerPup.id}>
                    <Link to={`/pups/${organizerPup.id}`}>
                      {organizerPup.name}
                    </Link>
                    <br></br>
                  </li>
                ))}
              </ul>
            </div>
          )}
          <div>
            <button onClick={this.handleAttendees}>
              {this.state.showAttendees === true ? (
                <p className="PupdateProfile-attendee">Hide Attendee List</p>
              ) : (
                <p className="PupdateProfile-attendee">View Attendee List</p>
              )}
            </button>
            <AttendeeList
              showAttendees={this.state.showAttendees}
              attendees={this.state.attendees}
            />
          </div>
          <br></br>
          {this.state.userPupdate === true ? (
            <div>
              <Link to={`/edit/pupdates/${this.state.pupdate.id}`}>
                <button className="PupdateProfile-edit">Edit Pupdate</button>
              </Link>
              <button onClick={this.handleDeleteRequest}>Delete Pupdate</button>
            </div>
          ) : this.state.userAttending === true ? (
            <div>
              <button onClick={this.handleRsvpNo}>
                I can no longer attend
              </button>
            </div>
          ) : (
            <div>
              <button onClick={this.handleRsvpYes}>I'll be there!</button>
            </div>
          )}
          <button onClick={this.handleBack}>Back to Pupdates</button>
        </section>
      </div>
    );
  }
}

export default PupdateProfile;
