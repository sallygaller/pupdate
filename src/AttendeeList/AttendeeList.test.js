import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import PupdateProfile from "../PupdateProfile/PupdateProfile";
import { pups, pupdates } from "../Utils/TestFiles";
import AttendeeList from "./AttendeeList";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <BrowserRouter>
      <AttendeeList attendees={[pups]} />
    </BrowserRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the UI as expected", () => {
  const tree = renderer
    .create(
      <PupdateProfile
        pupdate={pupdates[0]}
        match={{ params: { id: 1 }, isExact: true, path: "", url: "" }}
      >
        <AttendeeList name="AttendeeList" attendees={[pups]} />
      </PupdateProfile>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
