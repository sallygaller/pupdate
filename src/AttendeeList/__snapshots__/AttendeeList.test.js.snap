// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="PupdateProfile"
>
  <section>
    <h2>
      Pupdate on 
      March 21, 2021
    </h2>
    <div>
      <p>
        Time:
         
        Invalid date
         
        - 
        Invalid date
         
        <br />
        Location: 
      </p>
    </div>
    <div />
    <div>
      <h3>
        Organized by:
      </h3>
      <ul />
    </div>
    <div>
      <div
        className="AttendeeList-main"
      >
        <div
          className="AttendeeList-section"
        >
          <button>
            toggle
          </button>
          <div
            className="AttendeeList-sectionhead"
            onClick={[Function]}
          >
            <h3>
              Attendee List
            </h3>
          </div>
          <div
            className="AttendeeList-hidden"
          >
            <div
              className="AttendeeList-article"
            >
              <ul>
                <li>
                  No attendees yet!
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
    <br />
    <div>
      <button
        className="PupdateProfile-button"
        onClick={[Function]}
      >
        I'll be there!
      </button>
    </div>
    <button
      className="PupdateProfile-button"
      onClick={[Function]}
    >
      Back to Pupdates
    </button>
  </section>
</div>
`;
